libafl_libfuzzer/libafl_libfuzzer_runtime/src/feedbacks.rs

```diff
@@ -113,7 +113,7 @@ impl LibfuzzerCrashCauseFeedback {
 
 impl<S> Feedback<S> for LibfuzzerCrashCauseFeedback
 where
-    S: UsesInput<Input = BytesInput> + HasClientPerfMonitor,
```

Comment:
I don't understand why this code compiles...?
@addisoncrump 

Essentially we changed UsesInput to State.
But in UsesInput you can force(?) its `Input` to be `BytesInput`.
Now we're using `State` but it doesn't know anything about `Input` type.
Why this works????

Reply:
State<Input = BytesInput>

Reply:
ok. it just compiles..
but why syntax like 
```
State<Input = BytesInput>
```

Shouldn't this `Input` a associate type of `State`?
but the trait `State` is this:
```
pub trait State:
    UsesInput + Serialize + DeserializeOwned + MaybeHasClientPerfMonitor + MaybeHasScalabilityMonitor
{
}
```
And there's no such thing like `Input` in this `State` definition